packages/block-library/src/post-excerpt/index.php

```diff
@@ -45,7 +29,28 @@ function render_block_core_post_excerpt( $attributes, $content, $block ) {
 	 * `excerpt_more` filter and return nothing. This will
 	 * result in showing only one `read more` link at a time.
 	 */
+	$more_text           = ! empty( $attributes['moreText'] ) ? '<a class="wp-block-post-excerpt__more-link" href="' . esc_url( get_the_permalink( $block->context['postId'] ) ) . '">' . wp_kses_post( $attributes['moreText'] ) . '</a>' : '';
+	$filter_excerpt_more = static function ( $more ) use ( $more_text ) {
+		return empty( $more_text ) ? $more : '';
+	};
```

Comment:
In trunk, the `excerpt_more` filter is being added, but only after the call to `get_the_excerpt` (where the filter would be applied), so I think it's doing nothing. I've added a commit that also fixes that issue by moving where the filter is applied.